load/cliploader.py

```diff
@@ -0,0 +1,245 @@
+            logging.error("No meta data found for %s", metadata_filename)
+            return
+
+        metadata = tools.load_clip_metadata(metadata_filename)
+        valid_tracks = self._filter_clip_tracks(metadata)
+
+        if len(valid_tracks) == 0:
+            logging.error("No valid track data found for %s", filename)
+            return
+
+        clip = Clip(self.track_config)
+        clip.load_cptv(filename)
+        clip.parse_clip(metadata, self.config.loader.include_filtered_channel)
+
+        if self.track_config.enable_track_output:
+            self._export_tracks(filename, clip)
+
+        # write a preview
+        if self.previewer:
+            preview_filename = base_filename + "-preview" + ".mp4"
+            preview_filename = os.path.join(destination_folder, preview_filename)
+            self.previewer.create_individual_track_previews(preview_filename, clip)
+            self.previewer.export_clip_preview(preview_filename, clip)
+
+        if self.track_config.verbose:
+            num_frames = len(clip.frame_buffer.thermal)
+            ms_per_frame = (time.time() - start) * 1000 / max(1, num_frames)
+            self._log_message(
+                "Tracks {}.  Frames: {}, Took {:.1f}ms per frame".format(
+                    len(clip.tracks), num_frames, ms_per_frame
+                )
+            )
+
+    def _log_message(self, message):
+        """ Record message in stdout.  Will be printed if verbose is enabled. """
+        # note, python has really good logging... I should probably make use of this.
+        if self.track_config.verbose:
+            logging.info(message)
+
+    def _test_compression_time(self, full_path, clip):
+        for z in range(2):
+            start = time.time()
+            self.compression = tools.gzip_compression if z == 0 else None
+            for i in range(100):
+                clip._id = z * 1000 + i
+                self.database.create_clip(clip)
+
+                for track in clip.tracks:
+                    start_time, end_time = clip.start_and_end_time_absolute(
+                        track.start_s, track.end_s
+                    )
+
+                    self.database.add_track(
+                        clip.get_id(),
+                        track,
+                        opts=self.compression,
+                        start_time=start_time,
+                        end_time=end_time,
+                    )
+            end = time.time()
+            print(
+                "export tracks with compression {} took {}".format(
+                    self.compression, end - start
+                )
+            )
+
+
+def get_distributed_folder(name, num_folders=256, seed=31):
+    """Creates a hash of the name then returns the modulo num_folders"""
+    str_bytes = str.encode(name)
+    hash_code = 0
+    for byte in str_bytes:
+        hash_code = hash_code * seed + int(byte)
+
+    return str(hash_code % num_folders)
```

Comment:
How about returning this as a 2 character hex string so that all the directory names have the same length?
```python
return "{:02x}.format(hash_code % num_folders)
```